pyproject.toml

```diff
@@ -0,0 +1,98 @@
+[project]
+name = "firedrake"
+version = "0.14_dev"
+description = "An automated system for the portable solution of partial differential equations using the finite element method"
+readme = "README.rst"
+license = {file = "LICENSE"}
+authors = [
+  {name = "The Firedrake Team", email = "firedrake@imperial.ac.uk"},
+  {name = "Jack Betteridge", email = "J.Betteridge@imperial.ac.uk"},
```

Comment:
Think about contact.

Reply:
To not violate single source of truth we should somehow link this to https://github.com/firedrakeproject/firedrake/blob/master/docs/source/team.ini

Since we're not yet publishing this to PyPI I think this can be handled at a later point and just removed from this PR.

Reply:
Though the `team.ini` file distinguishes between further things like "active" and "inactive" core team members as well as having institutions and links. The `authors`/`maintainers` field in the `pyproject.toml` can only store names and emails.

It might be simplest just to have all of the current core developers listed as `maintainers` and to not have any `authors` specified.